edxlearndot/management/commands/update_learndot_enrolments.py

```diff
@@ -119,16 +122,16 @@ def handle(self, *args, **options):
                 # to explicitly update Learndot.
                 #
                 course_grade = CourseGradeFactory().read(enrollment.user, course)
-                if course_grade.passed and should_persist_grades(cm.edx_course_key):
+                if not course_grade:
```

Comment:
Hmm ok, i see what you mean about the `contact_id`..  I thought since it was cached, it wouldn't really matter if we're re-fetching, but ok.

But we shouldn't be calling `get_enrolment_id` if `contact_id is None` either, so could you catch that here too please?